### ISSUE TITLE
Request: Water entry sound. Bug: Ground footstep sound when walking in 1 node deep liquid, should be liquid footstep

### ISSUE THREAD

**Issue by Siirabs** [minetest/minetest#10527]
##### Minetest version

```
5.3.0
```

##### OS / Hardware

Operating system: Android 10

##### Summary

If you jump to water there ain't any sound effect. You will hear water sounds once you start moving in the water but not when you land in the water. 

Also if you run in 1 block deep water you only hear running sounds and none water splashing etc.
##### Steps to reproduce

1. Jump to water
2. Run in 1 block deep water

**Comment by paramat** [comment]
Feature request for liquid entry sounds and playing the sounds of the node the legs are in as higher priority than the node the feet are on. Both seem duplicates, i thought the second had a PR, will check.

Liquid sounds are implemented using footstep sounds, so are only triggered when moving in water.

**Comment by paramat** [comment]
First issue duplicate of #1369 (closed due to no core dev support, can reopen if there is).
Second issue duplicate of #4775
So, will close this in 1 month. If there is core dev support for a liquid entry sound please reopen #1369 and add the 'supported' label,

**Comment by Wuzzy2** [comment]
A water entry sounds seems like an obvious great feature to me. Many games that have water make a splash sound when you fall into water. Being completely silent for falling into water feels wrong.

But this issue probably needs reorganization.

**Comment by paramat** [comment]
Wuzzy2, a liquid entry sound is a nice idea of course, but here are some things we need to keep in mind:

It is a high-realism sound feature. There are 100s of other sound types of a similar realism level and we cannot have them all.
It would be bad design to have just one high-realism sound feature as it would be unbalanced with othr sound features.
The level of sound realism for MT is necessarily low, we need to keep all sound features balanced and near that level.
> Being completely silent for falling into water feels wrong.

Obviously, but MT sound is unrealistic in many other ways too. MT is a high-abstraction voxel game engine, it is not high-realism.
There are 100s of other aspects of sound we could describe as 'feeling wrong'.

Another point is that players rarely jump into liquid and then not move, they almost always swim, which triggers the water footstep sound. So usually there is a sound after a brief pause.

Keep in mind that in MTG, which is probably what the author is judging this on, has water footstep sounds with 1 silent sound, so occasionally there is a longer pause before sounds start. Perhaps this silent sound was not a good idea and should be removed?

I am not necessarily opposing a water entry sound, but am unsure it can be justified considering the above. It is certainly low priority.

**Comment by Desour** [comment]
This is not about realism. It's about sounds that sound good.

The water sounds in mtg do not sound like a water splash sound.

Argumenting against new features with the lack of features makes no sense.

>MT is a high-abstraction voxel game engine, it is not high-realism.

But the engine should provide features for the subgames so that they can be made better.

(Very realistic sounds might actually sound very good. (Eg. like the shoot sounds in openspades.) But that was not requested yet, so here we go: I would like to hear a splash sound, but when the head is under water the sounds should be damped. And there should be reverberation and everything. This is however not really related to this issue.)

**Comment by paramat** [comment]
It is about realism.
> The water sounds in mtg do not sound like a water splash sound.

Intentionally. They have to work for swimming on the surface and underwater, where a big splash is obviously unsuitable.
I chose and added those sounds.
The character of the sounds was carefully chosen for good reasons to work in multiple ways. So it is a splashy-sloshing that is a balance between a surface sound and an underwater sound, and a balance between a splash and a swimming slosh.
> Argumenting against new features with the lack of features

Not what i am doing, please reread my comments about sound realism. You are misrepresenting my comments.
> But the engine should provide features for the subgames so that they can be made better.

The MT engine does not support high realism, obviously. This is a cubic voxel game engine, it is inherently high-abstraction and low-realism.
The 'MT engine should do absolutely everything because it is a game engine' is an invalid argument.
> I would like to hear a splash sound, but when the head is under water the sounds should be damped. And there should be reverberation and everything.

That is unsuitably excessive for the abstraction and simplicity of MT.

**Comment by Desour** [comment]
It is not about realism. New features are not just to get more realism.

>>The water sounds in mtg do not sound like a water splash sound.
>
>Intentionally. They have to work for swimming on the surface and underwater, where a big splash is obviously unsuitable.

But a splash is needed. It is a feature that was already requested more than once.
For clarification: I've wrote that because you claimed that a separate sound isn't needed because of the swimming sound:
>Another point is that players rarely jump into liquid and then not move, they almost always swim, which triggers the water footstep sound.

I've just tested the water sounds in mtg again, and they do work fine for swimming, but not for diving (you can hear water surface sounds).
And water entry is super unsuitable anyways, as being silent.
It's not mtg's fault, that the sounds are bad. It's an engine limitation that was requested to be filled.

>Not what i am doing, please reread my comments about sound realism. You are misrepresenting my comments.

Well, you did write that minetest's level of sound features (you say they were only useful for uber-high realism) is low. And you did write that this level needs to be kept low (= don't add even more features) because it is low (= absence of features).

>The 'MT engine should do absolutely everything because it is a game engine' is an invalid argument.

I did not write this. It should not do everything. But it should allow to do many, and especially useful things. And users seem to think that liquid entry sounds are important and useful.

>>I would like to hear a splash sound, but when the head is under water the sounds should be damped. And there should be reverberation and everything.
>
>That is unsuitably excessive for the abstraction and simplicity of MT.

It is as excessive and suitable as nice graphic features, such as shadows or other cool effects.

I do not understand what is causing you to be against most new features. The features of a game are the reason I play it, so getting more and more excessive ones of them is something positive in my eyes.

**Comment by Wuzzy2** [comment]
I don't understand this train of thought at all. What possible harm is done if liquid entry sounds are added? Seriously, what's the harm?
Will it reduce performance? No.
Will it make the API notably more complex? No.
It is an obscure feature that nobody wants? No.
Will it force all games to now have a splash sound? No, obviously this is supposed to be optional, like any other sound.
Will it allow games to make the game feel a little bit more alive? YES!

And your decision to classify a sound as “high realism” seems VERY arbitrary. It is subjective, not objective. For example, why is swimming *not* “high realism”? With this argument you can really reject any sound, just on the basis on which sound you classify as “high realism”. So that's a weak argument.

> Another point is that players rarely jump into liquid and then not move, they almost always swim, which triggers the water footstep sound.

That's a pretty weak argument. You admitted that there will be no sound if the player falls into water and does not move. Even if this is rare, it's still a possibility. Also, this suggestion is NOT about MTG.

And the argument about realism does not make sense anyway. I do not care about realism. Sounds are about making the game feel more alive. Sounds should be played when stuff happens.

**Comment by sfan5** [comment]
Nevermind it makes more sense to reopen #1369, I'll do that.

**Comment by paramat** [comment]
Desour:
> But a splash is needed. It is a feature that was already requested more than once.

Being requested more than once or even many times does not necessarily make a feature suitable.
>  I've wrote that because you claimed that a separate sound isn't needed because of the swimming sound:

I did not claim it 'isn't needed'. I was explaining why the need for it is not as high as it might initially seem.
> And users seem to think that liquid entry sounds are important and useful.

A few users have, you cannot imply most users think it important.
> I do not understand what is causing you to be against most new features.

I am not, it is certainly far less than 50%.
And i am not 'against' this feature as you imply. I think it is borderline suitable, i am unsure, and i think there are some things to consider.
Here are some self-quotes from my earlier comments to prove you are greatly exaggerating and incorrect:
> a liquid entry sound is a nice idea of course, but here are some things we need to keep in mind:

> I am not necessarily opposing a water entry sound, but am unsure it can be justified considering the above. It is certainly low priority.

So please do not misrepresent me.


Wuzzy2,
> Also, this suggestion is NOT about MTG.

I am not talking about MTG. Any game that considers liquid sounds important will obviously already have water footsteps defined, so MTG or not is irrelevant.

**Comment by paramat** [comment]
I had some doubts, but i was thinking about this during the night and it makes sense to me now. It is consistent with the footstep sound played when a player lands on the ground.
Because i am now not happy with my comments in this issue and because they upset people i am deleting them =)
I will suggest some ideas in the original issue.
